File: robocode-tankroyale-bot-api-csharp/src/events/ConnectionErrorEvent.cs
using System;
using Newtonsoft.Json;

namespace Robocode.TankRoyale.BotApi
{
  /// <summary>
  /// Event occurring when a connection error occurs.
  /// </summary>
  public sealed class ConnectionErrorEvent
  {
    /// <summary>The exception causing the error.</summary>
    public Exception Exception { get; }

    /// <summary>
    /// Initializes a new instance of the ConnectionErrorEvent class.
    /// </summary>
    /// <param name="exception">The exception causing the error.</param>
    [JsonConstructor]
    public ConnectionErrorEvent(Exception exception) => Exception = exception;
  }
}

File: bot-api/python/src/robocode_tank_royale/bot_api/events/connection_error_event.py
from dataclasses import dataclass

from .connection_event import ConnectionEvent


@dataclass(frozen=True, repr=True)
class ConnectionErrorEvent(ConnectionEvent):
    """
    Represents an event that occurs when a connection error happens.

    Attributes:
        error (Exception): The exception representing the connection error.
    """

    error: Exception | None = None
